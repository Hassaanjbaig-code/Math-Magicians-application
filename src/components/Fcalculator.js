import React, { useState } from 'react';
import './Calculator.css';
import calculate from './logic/calculate';

const Fcalculator = () => {
  const [items, setitems] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const handleClick = (e) => {
    setitems(calculate(items, e.target.textContent));
  };
  const { total, operation, next } = items;
  return (
    <div className="calculator">
      <h3>Let&apos;s do some math!</h3>
      <ul className="main">
        <p className="text" type="text">
          <span>
            {total}
            {operation}
            {next}
          </span>
        </p>
        <li className="row">
          <button onClick={handleClick} type="button">AC</button>
          <button onClick={handleClick} type="button">+/-</button>
          <button onClick={handleClick} type="button">%</button>
          <button onClick={handleClick} type="button" className="right">&#247;</button>
        </li>
        <li className="row">
          <button onClick={handleClick} type="button">7</button>
          <button onClick={handleClick} type="button">8</button>
          <button onClick={handleClick} type="button">9</button>
          <button onClick={handleClick} type="button" className="right">x</button>
        </li>
        <li className="row">
          <button onClick={handleClick} type="button">4</button>
          <button onClick={handleClick} type="button">5</button>
          <button onClick={handleClick} type="button">6</button>
          <button onClick={handleClick} type="button" className="right">-</button>
        </li>
        <li className="row">
          <button onClick={handleClick} type="button">1</button>
          <button onClick={handleClick} type="button">2</button>
          <button onClick={handleClick} type="button">3</button>
          <button onClick={handleClick} type="button" className="right">+</button>
        </li>
        <li className="row">
          <button onClick={handleClick} type="button" className="zero">0</button>
          <button onClick={handleClick} type="button">.</button>
          <button onClick={handleClick} type="button" className="right">=</button>
        </li>
      </ul>
    </div>
  );
};

export default Fcalculator;
