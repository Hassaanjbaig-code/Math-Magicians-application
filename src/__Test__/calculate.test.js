import React from 'react';
import renderer from 'react-test-renderer';
import Fcalculator from '../components/Fcalculator';

it('matches snapshot', () => {
  const tree = renderer.create(<Fcalculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
