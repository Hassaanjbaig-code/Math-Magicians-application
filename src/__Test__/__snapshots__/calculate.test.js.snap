// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="calculator"
>
  <h3>
    Let's do some math!
  </h3>
  <ul
    className="main"
  >
    <p
      className="text"
      type="text"
    >
      <span />
    </p>
    <li
      className="row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="right"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </li>
    <li
      className="row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="right"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </li>
    <li
      className="row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="right"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </li>
    <li
      className="row"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="right"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </li>
    <li
      className="row"
    >
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="right"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </li>
  </ul>
</div>
`;
